wallet/wallet.go

```diff
@@ -33,9 +29,10 @@ type WalletProtocol struct {
 type CounterpartyType int
 
 const (
-	CounterpartyTypeAnyone CounterpartyType = 0
-	CounterpartyTypeSelf   CounterpartyType = 1
-	CounterpartyTypeOther  CounterpartyType = 2
+	CounterpartyUninitialized CounterpartyType = 0
+	CounterpartyTypeAnyone    CounterpartyType = 1
+	CounterpartyTypeSelf      CounterpartyType = 2
+	CounterpartyTypeOther     CounterpartyType = 3
```

Comment:
You could use the `iota` golang feature.

```suggestion
	CounterpartyUninitialized CounterpartyType = iota
	CounterpartyTypeAnyone    
	CounterpartyTypeSelf      
	CounterpartyTypeOther     
```

Reply:
Yes, this is a good case for using `iota`. Personally I prefer to be explicit to avoid any potential confusion. The `go-sdk` repo does use `iota` in other places, but I also see @shruggr being explicit, e.g. [here](https://github.com/b-open-io/go-sdk/blob/c31fbfb/overlay/types.go#L59).

Reply:
Sure 👍 The explicit way is sometimes more descriptive.